cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/KeyHelper.java

```diff
@@ -17,58 +17,65 @@
 package co.cask.cdap.data2.metadata.dataset;
 
 import co.cask.cdap.data2.dataset2.lib.table.MDSKey;
-import co.cask.cdap.proto.Id;
-import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.DatasetId;
+import co.cask.cdap.proto.id.NamespacedId;
+import co.cask.cdap.proto.id.ProgramId;
+import co.cask.cdap.proto.id.StreamId;
+import co.cask.cdap.proto.id.StreamViewId;
+
+import java.util.Arrays;
 
 /**
  * Helper methods for keys of {@link MetadataDataset}.
  */
 // Note: these methods were refactored from MetadataDataset class. Once CDAP-3657 is fixed, these methods will need
 // to be cleaned up CDAP-4291
 final class KeyHelper {
-  static void addTargetIdToKey(MDSKey.Builder builder, Id.NamespacedId namespacedId) {
+  static void addTargetIdToKey(MDSKey.Builder builder, NamespacedId namespacedId) {
     String type = getTargetType(namespacedId);
-    if (type.equals(Id.Program.class.getSimpleName())) {
-      Id.Program program = (Id.Program) namespacedId;
-      String namespaceId = program.getNamespaceId();
-      String appId = program.getApplicationId();
+    if (type.equals(ProgramId.class.getSimpleName())) {
+      ProgramId program = (ProgramId) namespacedId;
+      String namespaceId = program.getNamespace();
+      String appId = program.getApplication();
       String programType = program.getType().name();
-      String programId = program.getId();
+      String programId = program.getProgram();
       builder.add(namespaceId);
       builder.add(appId);
       builder.add(programType);
       builder.add(programId);
-    } else if (type.equals(Id.Application.class.getSimpleName())) {
-      Id.Application application = (Id.Application) namespacedId;
-      String namespaceId = application.getNamespaceId();
-      String instanceId = application.getId();
+    } else if (type.equals(ApplicationId.class.getSimpleName())) {
+      ApplicationId application = (ApplicationId) namespacedId;
+      String namespaceId = application.getNamespace();
+      String instanceId = application.getApplication();
```

Comment:
this is existing code, but wonder why this is called ``instanceId`` - change it to ``appId``?